File: is_match.java
public class Main {
    public static boolean isMatch(String s, String p) {
        int m = s.length(), n = p.length();
        boolean[][] dp = new boolean[m + 1][n + 1];
        dp[m][n] = true;

        for (int i = m; i >= 0; --i) {
            for (int j = n - 1; j >= 0; --j) {
                boolean first_match = i < m && (s.charAt(i) == p.charAt(j) || p.charAt(j) == '.');
                if (j + 1 < n && p.charAt(j + 1) == '*') {
                    dp[i][j] = dp[i][j + 2] || (first_match && dp[i + 1][j]);
                } else {
                    dp[i][j] = first_match && dp[i + 1][j + 1];
                }
            }
        }
        return dp[0][0];
    }

    public static void main(String[] args) {
        String s = "aaaaaaaaaaaaaaaaaaa";
        String p = "a*a*a*a*a*a*a*a*a*b";
        System.out.println(isMatch(s, p)); // Output: false
    }
}

File: is_match.dart
bool isMatch(String s, String p) {
  int m = s.length, n = p.length;
  List<List<bool>> dp = List.generate(m + 1, (_) => List<bool>.filled(n + 1, false));
  dp[m][n] = true;

  for (int i = m; i >= 0; i--) {
    for (int j = n - 1; j >= 0; j--) {
      bool firstMatch = i < m && (s[i] == p[j] || p[j] == '.');
      if (j + 1 < n && p[j + 1] == '*') {
        dp[i][j] = dp[i][j + 2] || (firstMatch && dp[i + 1][j]);
      } else {
        dp[i][j] = firstMatch && dp[i + 1][j + 1];
      }
    }
  }
  return dp[0][0];
}

void main() {
  String s = "aaaaaaaaaaaaaaaaaaa";
  String p = "a*a*a*a*a*a*a*a*a*b";
  print(isMatch(s, p)); // Output: false
}
